packages/query-core/src/tests/queriesObserver.test.tsx

```diff
@@ -50,11 +50,11 @@ describe('queriesObserver', () => {
     unsubscribe()
     expect(observerResult).toMatchObject([{ data: 1 }, { data: 2 }])
 
-    expect(mockLogger.error).toHaveBeenCalledTimes(2)
-    expect(mockLogger.error).toHaveBeenCalledWith(
+    expect(console.error).toHaveBeenCalledTimes(2)
+    expect(console.error).toHaveBeenCalledWith(
       'Passing a custom logger has been deprecated and will be removed in the next major version.',
     )
-    expect(mockLogger.error).toHaveBeenCalledWith(
+    expect(console.error).toHaveBeenCalledWith(
       'Passing a custom logger has been deprecated and will be removed in the next major version.',
```

Comment:
I think we have to put a spy on the console and then check if his has been called. We don't want the console to be spammed with errors during tests, as this will make it harder to find actual errors.

we have something similar e.g. here:

https://github.com/TanStack/query/blob/c376158afaddf7cd92a85491c3f18c1b123efe4b/packages/react-query/src/__tests__/ssr-hydration.test.tsx#L58-L59

At the end of the test, mocks need to be restored:

https://github.com/TanStack/query/blob/c376158afaddf7cd92a85491c3f18c1b123efe4b/packages/react-query/src/__tests__/ssr-hydration.test.tsx#L134

this would be needed for all tests please

Reply:
I have made the Recommended Changes and implemented console mock everywhere a console.error call was made in tests.
somehow some tests are failing and the console output is a mess

![image](https://user-images.githubusercontent.com/1729481/209934201-98b82a00-c1ee-45a4-a767-4471010fc914.png)

Reply:
I think we're still missing this from the issue description:

> do not log failed queries to the console anymore, as failed queries are not a programmer "error" that we need to point people towards. The network tab shows failed requests just fine.

Reply:
This will also make it easier for people to test erroneous queries because they don't need to turn off the logger anymore - and we also don't need to do that internally then for every failed query :)

Reply:
got it 😅, will do the required changes, thanks for being patient.